Update index.rst

Bowrna opened the pull request:
Mounting config in docker way of running airflow

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

Bowrna committed:
Update index.rst

Mounting config in docker way of running airflow

Bowrna committed:
Merge branch 'main' into patch-3

Bowrna committed:
Merge branch 'main' into patch-3

Bowrna committed:
Merge branch 'main' into patch-3

potiuk commented on docs/apache-airflow/howto/docker-compose/index.rst:
```diff
@@ -101,6 +101,7 @@ Some directories in the container are mounted, which means that their contents a
 
 - ``./dags`` - you can put your DAG files here.
 - ``./logs`` - contains logs from task execution and scheduler.
+- ``./config`` - you can add custom log parser or add ``airflow_local_settings.py`` to configure cluster policy.
```
Hm. But I believe config is not mounted :

```
  volumes:
    - ${AIRFLOW_PROJ_DIR:-.}/dags:/opt/airflow/dags
    - ${AIRFLOW_PROJ_DIR:-.}/logs:/opt/airflow/logs
    - ${AIRFLOW_PROJ_DIR:-.}/plugins:/opt/airflow/plugins
```

Bowrna replied:
oops i thought its missed out in documentation part alone. could i add that as part of the docker-compose.yml file too @potiuk ?

Bowrna replied:
@potiuk could you let me know if i can add as part of docker-compose.yml? bringing this up as a gentle reminder

pankajastro replied:
@potiuk I think it mounted here https://github.com/apache/airflow/blob/90ee7639478bc8534164c4bbb05a1ccfb04272dd/docs/apache-airflow/howto/docker-compose/docker-compose.yaml#L75

or do we need to mount it at some other places too?

potiuk replied:
config?  @pankajastro ?

potiuk replied:
> @potiuk could you let me know if i can add as part of docker-compose.yml? bringing this up as a gentle reminder

Yes. you can.

pankajastro replied:
> config? @pankajastro ?

yeah it missing 🤦‍♂️

hussein-awala committed:
Add a new parameter for base sensor to catch the exceptions in poke method (#30293)

* add a new parameter for base sensor to catch the exception in poke method

* add unit test for soft_fail parameter

dolfinus committed:
Allow to set limits for XCOM container (#28125)

rajaths010494 committed:
Add AWS deferrable BatchOperator (#29300)

This PR donates the following BatchOperator deferrable developed in [astronomer-providers](https://github.com/astronomer/astronomer-providers) repo to apache airflow.

okue committed:
Update dead link in Sentry integration document (#30486)

* Update dead link in Sentry integration document

* fix

vandonr-amz committed:
Add more info to quicksight error messages (#30466)

potiuk committed:
Revert "Add AWS deferrable BatchOperator (#29300)" (#30489)

This reverts commit 77c272e6e8ecda0ce48917064e58ba14f6a15844.

potiuk committed:
Fix output to outputs typos in ci.yaml everywhere (#30490)

(Facepalm) The typo of output -> outputs from #30482  was also in the
ci.yaml where it was used and it was missed in this PR.

I can blame the GitHub Actions stupid choice of accepting typoed
names of outputs and replacing them with blank strings (which I
raised as an issue a long time ago)

dnskr committed:
Reformat chart templates part 3 (#30312)

potiuk committed:
Move Pydantic classes for ORM objects to serialization (#30484)

The Pydantic classes are really part of the serialization
mechanism and they should be moved there, rather than kept in
the core packages they serialize, following our serialization
approach.

potiuk committed:
Separate mypy pre-commit checks (#30502)

Previously all mypy pre-commit checks were run as one "run-mypy"
check, but that does not allow to run them separately when trying
to fix some of them, only for a specific part of the sources.

This PR splits them into "dev", "core", "providers" and "docs".

hussein-awala committed:
Avoid logging sensitive information in triggerer job log (#30110)

* Change trigger name to task id instead of repr(trigger) to avoid logging sensitive information

ashb committed:
Allow specifying a `max_depth` to the `redact()` call (#30505)

The default was hard-coded as 5 which is suitable for the logs
redacting, but the OpenLineage PR would like to be able to use a deeper
depth.

jedcunningham committed:
Fix deprecation warning in `example_sensor_decorator` DAG (#30513)

potiuk committed:
Put AIP-44 internal API behind feature flag (#30510)

This includes:

* configurable setting with defaults taken from env variable
* raising exception if config variables are used with feature
  flag not enabled
* hiding config values (adding mechanism to hide config values
  that are set for the future versions)
* skipping tests

potiuk committed:
Summarize skipped tests after tests are run (#30520)

When Pytest run tests it provides a summary of the tests. We are
running a lot of the tests so we are really interested only in cases
that are "interesting". So far we were not showing "skipped" tests
in the summary, because there were cases where a lot of tests
were skipped (mostly when integration tests were run - we collected
tests from "tests" folder and run only those tests that were not
skipped by @integration mark.

This however changed in #28170 as we moved all integration
tests to "integration" subfolder and now instead of large number of
skipped tests we run them selectively for each integration.

This should help in verifying that the skipped tests were skipped
for a good reason (and that we actually see which tests have been
skipped).

uranusjr committed:
Add more type hints to the code base (#30503)

* Fully type Pool

Also fix a bug where create_or_update_pool silently fails when an empty
name is given. An error is raised instead now.

* Add types to 'airflow dags'

* Add types to 'airflow task' and 'airflow job'

* Improve KubernetesExecutor typing

* Add types to BackfillJob

This triggers an existing typing bug that pickle_id is incorrectly typed
as str in executors, while it should be int in practice. This is fixed
to keep things straight.

* Add types to job classes

* Fix missing DagModel case in SchedulerJob

* Add types to DagCode

* Add more types to DagRun

* Add types to serialized DAG model

* Add more types to TaskInstance and TaskReschedule

* Add types to Trigger

* Add types to MetastoreBackend

* Add types to external task sensor

* Add types to AirflowSecurityManager

This uncovers a couple of incorrect type hints in the base
SecurityManager (in fab_security), which are also fixed.

* Add types to views

This slightly improves how view functions are typechecked and should
prevent some trivial bugs.

o-nikolas committed:
Type related import optimization for Executors (#30361)

Move some expensive typing related imports to be under TYPE_CHECKING

utkarsharma2 committed:
Fix link to pre-commit-hook section (#30522)

* Change static link

* Update LOCAL_VIRTUALENV.rst

pierrejeambrun committed:
Do not use template literals to construct html elements (#30447)

potiuk committed:
Enable AIP-44 and AIP-52 by default for development and CI on main (#30521)

* Enable AIP-44 and AIP-52 by default for development and CI on main

The AIP-44 and AIP-52 are controlled now by environment variables,
however those variables were not passed by default to inside the
docker-compose environment so they had no effect when set on the
ci.yaml. This PR fixes it, but it also sets the variables to
be enabled by default in Breeze environment and when the tests
are run locally in main using local venv so that the contributors
are not surprised when they try to reproduce local failures.

In 2.6 branch, we will set both variables to "false" by default
in ci.yml, so that the tests are not run when we cherry-pick the changes.

* Update scripts/ci/docker-compose/devcontainer.env

potiuk committed:
Run "api_internal" tests in CI (#30518)

* Run "api_internal" tests in CI

While adding a feature flag for AIP-44 I noticed that due to a
weird naming we have in tests, the "api_internal" tests were
actually excluded from running - this was due to a combination of
factors:

* When API tests are are run, only "api" and "api_connexion" were
  added to API_tests
* We already have "api" folder in "tests" (for experimental api)
* finding "Other" tests should cover it but it exluded "api" tests but
  the way it is implemented, it took the "api" prefix and excluded
  all the test directories that were starting with "api" (including
  "api_internal/*" ones

This change addresses it twofold:

* The "api_internal" tests are added explicitly to the "API"
  test type
* The "tests/api" folder with tests for the experimental API
  has been renamed to "api_experimental" (including integration
  tests)

This should set the "internal_api" tests to run in the API test
type, and renaming the "api" to api_experimental should avoid
accidental skipping the tests in case someone adds
"tests/api_SOMETHING" in the future.

* Update Dockerfile.ci

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

* Update entrypoint_ci.sh

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

vanducng committed:
`EmailOperator`: fix wrong assignment of `from_email` (#30524)

* `EmailOperator`: fix wrong assignment of `from_email`

ashb committed:
Add asgiref as a core dependency (#30527)

We added asgiref to core a few months back for the `sync_to_async` in
airflow.triggers.external_task.

Although the core http provider depends on asgiref since v 4.2 it is
possible to have an older version http installed meaning that you end up
without asgiref, which leads to every dag failing to parse as the
"dependency detector" code inside the DAG Serializer ends up importing
this module!

eladkal committed:
improve first PR bot comment (#30529)

jedcunningham committed:
Put AIP-52 setup/teardown tasks behind feature flag (#30509)

We aren't going to land AIP-52 in time for 2.6, so put the authoring api
behind a feature flag. I've chosen to put it in `airflow.settings` so
users can set it in `airflow_local_settings`, or set it via env var.

vincbeck committed:
Add tests to PythonOperator (#30362)

* Add tests to airflow/operators/python.py

* Convert log error of _BasePythonVirtualenvOperator._read_result() into a custom exception class

* Improve deserialization error handling

---------

Co-authored-by: Shahar Epstein <shahar1@live.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

AetherUnbound committed:
Make AWS RDS operator page title consistent (#30536)

This PR changes the title of the RDS documentation page from "Amazon Relational Database Service Documentation (RDS)" to "Amazon Relational Database Service (RDS)". This page was the only one with the word "Documentation" in its title, and several other services had a similar title format of ("Amazon <full service name> (<acronym>)"), for example ["Amazon Simple Notification Service (SNS)"](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/operators/sns.html).

potiuk committed:
Revert "Add tests to PythonOperator (#30362)" (#30540)

This reverts commit b4f3efd36a0566ef9d34baf071d935c0655a02ef.

potiuk committed:
Add new known warnings after dependencies upgrade. (#30539)

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

hussein-awala committed:
Fix AzureDataFactoryPipelineRunLink get_link method (#30514)

Commit:
Use default connection id for KubernetesPodOperator (#28848)

* Use default connection id for KubernetesPodOperator

---------

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

potiuk committed:
Fix dynamic imports in google ads vendored in library (#30544)

This is a further fix to vendor-in google-ads library:

* moving from _vendor to google_vendor is needed to keep
  the tree structure separate from `google` package used
  in google provider. If we do not do that, ads library
  will find "google" package when walking up the tree
  and will determine that this is the "top" google package

* dynamic imports of ads library imports have been updated
  to also import from the vendored-in library

* only v12 version is supported

Closes: #30526

vchiapaikeo committed:
BigQueryHook list_rows/get_datasets_list can return iterator (#30543)

VladaZakharova committed:
Add deferrable mode to GKEStartPodOperator (#29266)

* Add deferrable mode to GKEStartPodOperator

* Change naming for GKEHook and add comments

* Rebase main, revert unrelated changes

* Add review suggestions + rebase

* Add deprecation warning for deleted method + rebase

hussein-awala committed:
Accept None for `EmailOperator.from_email` to load it from smtp connection (#30533)

* Add default None for  to load it from smtp connection

Commit:
Update DV360 operators to use API v2 (#30326)

* Update DV360 operators to use API v2

* Update display_video.rst

* fixup! Update display_video.rst

* fixup! Update display_video.rst

---------

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

eladkal committed:
Prepare docs for ad hoc release of Providers (#30545)

* Prepare docs for ad hoc release of Providers

* add smtp provider

* add google

potiuk committed:
Add --one-pass-only parameter to breeze docs build command (#30555)

The parameter was previously supported in the docs-build script
but it was not exposed via breeze commands.

It allows to iterate faster on docs building, by default docs building
runs up to 3 passes in order to account for new cross-references
between multiple providers, this flag makes it one pass, which makes
it faster to summarize the errors when you try to nail down a problem
with docs.

hussein-awala committed:
Load subscription_id from extra__azure__subscriptionId (#30556)

dinigo committed:
Use the engine provided in the session (#29804)

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

eladkal committed:
fix release doc for providers (#30559)

jedcunningham committed:
Change timer type back to float (#30532)

potiuk committed:
Decouple "job runner" from BaseJob ORM model (#30255)

* Decouple "job runner" from BaseJob ORM model

Originally BaseJob ORM model was extended and Polymorphism has been
used to tie different execution logic to different job types. This
has proven to be difficult to handle during AIP-44 implementation
(internal API) because LocalTaskJob, DagProcessorJob and TriggererJob
are all going to not use the ORM BaseJob model, but they should use
BaseJobPydantic instead. In order to make it possible, we introduce
a new type of object BaseJobRunner and make BaseJob use the runners
instead.

This way, the BaseJobRunners are used for the logic of each of the
job, where single, non-polimorphic BaseJob is used to keep the
records in the database - as a follow up it will allow to completely
decouple the job database operations and move it to internal_api
component when db-lesss mode is enabled.

Closes: #30294

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

potiuk committed:
Fix one more dynamic import needed for vendored-in google ads (#30564)

Continuation of #30544

potiuk committed:
Move Pydantic class serialization under AIP-44 feature flag (#30560)

The Pydantic representation of the ORM models is only used
in AIP-44 in-progress feature, and we are moving to a new
seialization implementation (more modular) in a near feature
so in order to not unecessarily extend features in old
serialization, but allow to test AIP-44, we are moving the
use_pydantic_models parameter and it's implementation under
_ENABLE_AIP_44 feature flag, so that it is not used accidentally.

We will eventually remove it and add Pydantic serialization to
the new serialization implementation.

dim-ops committed:
Add podAnnotations to PgBouncer (#30168)

elongl committed:
Added support for using SHA digest of Docker images (#30214)

pierrejeambrun committed:
Bump json5 to 1.0.2 and eslint-plugin-import to 2.27.5 in /airflow/www (#30568)

Bumping from 1.0.1 for json5 and 2.26.0

prakash12 committed:
Update dataproc.rst (#30566)

Making statement more contextual so change proposed here is "a provide" to "to provide"

uranusjr committed:
Quieter output during asset compilation (#30565)

The "Still waiting ....." message was emitted every second, which can be
quite noisy even on moderate machines. This reduces the message to once
every 5 seconds.

potiuk committed:
Rename JobRunner modules to *_job_runner and base_job* to job (#30302)

The #30255 introduced "JobRunner" concept and decoupled the job logic
from the ORM polymorphic *Job objects. The change was implemented
in the way to minimise the review effort needed, so it avoided renaming
the modules for the runners (from `_job` to `_job_runner`).

Also BaseJob lost its "polymorphism" properties so the package, and class name
can be renamed to simply job.

This PR completes the JobRunner concept introduction by applying the
renames.

Closes: #30296

Commit:
Speed up dag runs deletion (#30330)

* Provide custom deletion for dag runs to speed up when a dag run has a lot of related task instances
---------

Co-authored-by: Zhyhimont Dmitry <zhyhimont.d@profitero.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

erdos2n committed:
Adding taskflow API example for sensors (#30344)

feluelle committed:
Use connection URI in SqliteHook (#28721)

* Use connection URI in SqliteHook

This allows the user to define more sqlite args such as mode. See https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#uri-connections for details.
- remove unsupported schema, login and password fields in docs
- add info about host field to docs

jedcunningham committed:
Release notes for helm chart 1.9.0 (#30570)

potiuk committed:
Do not remove docker provider for Airflow 2.3 check (#30483)

This removal is a remnant of old docker provider for 2.2 and should
not be happening.

potiuk committed:
Separate and split run job method into prepare/execute/complete steps (#30308)

* Separate and split run job method into prepare/execute/complete steps

As a follow-up after decoupling of the job logic from the BaseJob
ORM object (#30255), the `run` method of BaseJob should also be
decoupled from it (allowing BaseJobPydantic to be passed) as well
as split into three steps, in order to allow db-less mode.

The "prepare" and "complete" steps of the `run` method are modifying
BaseJob ORM-mapped object, so they should be called over the
internal-api from LocalTask, DafFileProcessor and Triggerer running
in db-less mode. The "execute" method however does not need the
database however and should be run locally.

This is not yet full AIP-44 conversion, this is a prerequisite to do
so - and AIP-44 conversion will be done as a follow-up after this one.

However we added a mermaid diagram showing the job lifecycle with and
without Internal API to make it easier to reason about it

Closes: #30295

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

jedcunningham committed:
Update SQLAlchemy `select()` to new style (#30515)

SQLAlchemy has a new style for `select() that is standard for 2.0. This
updates our uses of it to avoid `RemovedIn20Warning` warnings.

https://docs.sqlalchemy.org/en/20/errors.html#select-construct-created-in-legacy-mode-keyword-arguments-etc

potiuk committed:
Remove JobRunners back reference from Job (#30376)

This is the final step of decoupling of the job runner from ORM
based BaseJob. After this change, finally we rich the state that
the BaseJob is just a state of the Job being run, but all
the logic is kept in separate "JobRunner" entity which just
keeps the reference to the job. Also it makes sure that
job in each runner is defined as appropriate for each job type:

* SchedulerJobRunner, BackfillJobRunner can only use BaseJob
* DagProcessorJobRunner, TriggererJobRunner and especially the
  LocalTaskJobRunner can keep both BaseJob and it's Pydantic
  BaseJobPydantic representation - for AIP-44 usage.

The highlights of this change:

* Job does not have job_runner reference any more
* Job is a mandatory parameter when creating each JobRunner
* run_job method takes as parameter the job (i.e. where the state
  of the job is called) and executor_callable - i.e. the method
  to run when the job gets executed
* heartbeat callback is also passed a generic callable in order
  to execute the post-heartbeat operation of each of the job
  type
* there is no more need to specify job_type when you create
  BaseJob, the job gets its type by a simply creating a runner
  with the job

This is the final stage of refactoring that was split into
reviewable stages: #30255 -> #30302 -> #30308 -> this PR.

Closes: #30325

jedcunningham committed:
Cast binding +1 in helm chart release vote email (#30590)

We will assume that the release manager for the helm chart wants to cast
a binding +1 vote :)

harishkesavarao committed:
Databricks SQL sensor (#30477)

* Renamed example DAG

hussein-awala committed:
Add Hussein to committers (#30589)

vandonr-amz committed:
Add support in AWS Batch Operator for multinode jobs (#29522)

picking up #28321 after it's been somewhat abandoned by the original author.
Addressed my own comment about empty array, and it should be good to go I think.

Initial description from @camilleanne:

Adds support for AWS Batch multinode jobs by allowing a node_overrides json object to be passed through to the boto3 submit_job method.

Adds support for multinode jobs by properly parsing the output of describe_jobs (which is different for container vs multinode) to extract the log stream name.
closes: #25522

namsick96 committed:
Fix CONTRIBUTORS_QUICK_START Doc (#30549)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

uranusjr committed:
Use custom validator for OpenAPI request body (#30596)

* Use custom validator for OpenAPI request body

The default error message for an empty request body from Connexion
is quite unhelpful (taken directly from JSONSchema). This custom
validator emits a more helpful message for this particular context.

* Add test for custom request body validator

Co-Authored-By: maahir22 <56473490+maahir22@users.noreply.github.com>

---------

Co-authored-by: maahir22 <56473490+maahir22@users.noreply.github.com>

uranusjr committed:
Remove 'run-' prefix from pre-commit jobs (#30597)

* Remove 'run-' prefix from pre-commit jobs

The job ID already implies 'run', and having the additional prefix
results in weird CLI, e.g. 'pre-commit run run-mypy-core'. This changes
the CLI to 'pre-commit run mypy-core', which reads better.

* Fix table marker

* Fix outdated pre-commit hook ID references

vincbeck committed:
Add ability to override waiter delay in EcsRunTaskOperator (#30586)

eladkal committed:
Prepare docs for RC2 of provider wave (#30606)

luos-fc committed:
Deactivate DAGs deleted from within zipfiles (#30608)

dstaple committed:
DagBag: Use dag.fileloc instead of dag.full_filepath in exception message (#30610)

Co-authored-by: Douglas Staple <staple.douglas@gmail.com>

vincbeck committed:
Remove gauge scheduler.tasks.running (#30374)

* Remove gauge scheduler.tasks.running

* Add significant.rst file

* Update newsfragments/30374.significant.rst

---------

Co-authored-by: Niko Oliveira <onikolas@amazon.com>

hussein-awala committed:
Recover from `too old resource version exception` by retrieving the latest `resource_version` (#30425)

* Recover from `too old resource version exception` by retrieving the latest `resource_version`

* Update airflow/executors/kubernetes_executor.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

error418 committed:
docs: use correct import path for Dataset (#30617)

MarosMK committed:
Speed up TaskGroups with caching property of group_id (#30284)

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

dylanbstorey committed:
Fix `TriggerDagRunOperator` with deferrable parameter (#30406)

* readding after borked it

* pre-commit

* finally fixing after the github issue last week

* push fix

* feedback from hussein

potiuk committed:
Fix failing SQS tests on moto upgrade (#30625)

The new moto (4.1.7) performs additional validation on the queues
created during tests and it failes the tests when content
deduplication is not specified.

Explicit setting the deduplication mode, fixes the problem and
allows the new moto to be installed.

blinkseb committed:
fix possible race condition when refreshing DAGs (#30392)

* fix possible race condition when refreshing DAGs

* merge the two queries into one

* Remove provide_session from internal function

Since get_latest_version_hash_and_updated_datetime is internal and we
always pass in the session anyway, the provide_session decorator is
redundant and only introduce possibility for developer errors.

---------

Co-authored-by: Sébastien Brochet <sebastien.brochet@nielsen.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

jedcunningham committed:
Remove Norm and Hussein from the triage group (#30627)

Hussein is now a commiter and Norm has completed building out the
initial AIP-52 tasks.

moiseenkov committed:
Remove mysql-connector-python (#30487)

* Turn the package 'mysql-connector-python' as an optional feature

* Update airflow/providers/mysql/provider.yaml

* Update airflow/providers/mysql/CHANGELOG.rst

---------

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

potiuk committed:
Better error message where non-compatible providers are not excluded (#30629)

When compatibility check is performed for old version of Airflow,
we attempt to install all providers for the old version. However if
one of the providers has >= limit on Airflow for newer version of
Airflow, this installation lead to attempting to upgrade airflow
rather than fail, which could lead to misleading errors.

This change adds "airflow==x.y.z" taken from --use-airflow-version
flag to the `pip install` command, which should in this case fail
with much more accurate message, that the provider conflicts with
airflo version.

shanecglass committed:
Updating the links to the Dataform product documentation to fix 404 redirect error (#30631)

Updating the links to the Dataform product documentation to fix 404 redirect error

mrichman committed:
New AWS sensor — DynamoDBValueSensor (#28338)

jlaneve committed:
Remove duplicate param docstring in EksPodOperator (#30634)

oboki committed:
In `DockerOperator`, adding an attribute `tls_verify` to choose whether to validate certificate (#30309) (#30310)

* add `tls_verify` to choose whether to validate certificate (#30309)

---------

Co-authored-by: Hussein Awala <hussein@awala.fr>

hussein-awala committed:
Add `max_active_tis_per_dagrun` for Dynamic Task Mapping (#29094)

* add max_active_tis_per_dagrun param to BaseOperator

* set has_task_concurrency_limits when max_active_tis_per_dagrun is not None

* check if max_active_tis_per_dagrun is reached in the task deps

* check if all the tasks have None max_active_tis_per_dagrun before auto schedule the dagrun

* check if the max_active_tis_per_dagrun is reached before queuing the ti

* check max_active_tis_per_dagrun in backfill job

* fix current tests and ensure everything is ok before adding new tests

* refacto TestTaskConcurrencyDep

* fix a bug in TaskConcurrencyDep

* test max_active_tis_per_dagrun in TaskConcurrencyDep

* tests max_active_tis_per_dagrun in TestTaskInstance

* test dag_file_processor with max_active_tis_per_dagrun

* test scheduling with max_active_tis_per_dagrun on different DAG runs

* test scheduling mapped task with max_active_tis_per_dagrun

* test max_active_tis_per_dagrun with backfill CLI

* add new starved_tasks filter to avoid affecting the scheduling perf

* unify the usage of TaskInstance filters and use TI

* refacto concurrecy map type and create a new dataclass

* move docstring to ConcurrencyMap class and create a method for default_factory

* move concurrency_map creation to ConcurrencyMap class

* replace default dicts by counters

* replace all default dicts by counters in the scheduler_job_runner module

* suggestions from review

RNHTTR committed:
Simplify logic to resolve tasks stuck in queued despite stalled_task_timeout (#30375)

* simplify and consolidate logic for tasks stuck in queued

* simplify and consolidate logic for tasks stuck in queued

* simplify and consolidate logic for tasks stuck in queued

* fixed tests; updated fail stuck tasks to use run_with_db_retries

* mypy; fixed tests

* fix task_adoption_timeout in celery integration test

* addressing comments

* remove useless print

* fix typo

* move failure logic to executor

* fix scheduler job test

* adjustments for new scheduler job

* appeasing static checks

* fix test for new scheduler job paradigm

* Updating docs for deprecations

* news & small changes

* news & small changes

* Update newsfragments/30375.significant.rst

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Update newsfragments/30375.significant.rst

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* added cleanup stuck task functionality to base executor

* fix sloppy mistakes & mypy

* removing self.fail from base_executor

* Update airflow/jobs/scheduler_job_runner.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Update airflow/jobs/scheduler_job_runner.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Fix job_id filter

* Don't even run query if executor doesn't support timing out queued tasks

* Add support for LocalKubernetesExecutor and CeleryKubernetesExecutor

* Add config option to control how often it runs - we want it quicker than
the timeout

* Fixup newsfragment

* mark old KE pending pod check interval as deprecated by new check interval

* Fixup deprecation warnings

This more closely mirrors how deprecations are raised for "normal"
deprecations.

I've removed the depth, as moving up the stack doesn't really help the
user at all in this situation.

* Another deprecation cleanup

* Remove db retries

* Fix test

---------

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
Co-authored-by: Jed Cunningham <jedcunningham@apache.org>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

Commit:
Display Video 360 cleanup v1 API usage (#30577)

* Display Video 360 cleanup v1 API usage

* Update docs

hussein-awala committed:
Fix mapped tasks partial arguments when DAG default args are provided (#29913)

* Add a failing test to make it pass

* use partial_kwargs when they are provide and override only None values by dag default values

* update the test and check if the values are filled in the right order

* fix overriding retry_delay with default value when it is equal to 0

* add missing default value for inlets and outlets

* set partial_kwargs dict type to dict[str, Any] and remove type ignore comments

* create a dict for default values and use NotSet instead of None to support None as accepted value

* update partial typing by removing None type from some args and set NotSet for all args

* Tweak kwarg merging slightly

This should improve iteration a bit, I think.

* Fix unit tests

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

mobuchowski committed:
First commit of OpenLineage provider. (#29940)

This PR consistent mostly of code that was created in OpenLineage project. It 
consists of

- Provider wiring
- OpenLineageListener that uses Listener API to get notification about changes
  to TaskInstance and Dag states
- Extractor framework, which is used to extract lineage information from
  particular operators. It's ment to be replaced by direct implementation of
  lineage features in later phase and extracting them using DefaultExtractor.
  This PR does not include actual extractors, but code around using and registering them.
- OpenLineageAdapter that translates extracted information to OpenLineage events.
- Utils around specific Airflow OL facets and features

This is a base implementation that's not ment to be released yet, but to add
code modified to be consistent with Airflow standards, get early feedback and
provide canvas to add later features, docs, tests on.

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

ephraimbuddy committed:
Add v2-6-test and v2-6-stable to codecov and protected branches (#30640)

amoghrajesh committed:
Adding configuration to control retry parameters for k8s api client (#29809)

* Adding configuration to control retry parameters for k8s api client

* Handling review comments

* Fixing code bug

* Fixing failing tests

* Temporary commit with UT wip

* Fixing unit test

* Fixing the strict checks

* Handling review comments from Hussein

* Revert "Handling review comments from Hussein"

This reverts commit fa3bc260f7462c42620f694ee97b7f15c0b0b9c3.

* Fixing failing ut

* Reverting bad hack

* Updating logic in kube_client.py

Co-authored-by: Hussein Awala <hussein@awala.fr>

* Fixing unit tests

* Fixing unit tests

* Handling review comments from Ash

* Fix loading mock call args for python3.7

* Apply suggestions from code review

* fix static check

* add in 2.6.0

---------

Co-authored-by: Amogh <adesai@cloudera.com>
Co-authored-by: Hussein Awala <houssein.awala.96@gmail.com>

mikaeld committed:
fix(chart): webserver probes timeout and period. (#30609)

* fix(chart): webserver probes timeout and period

* Update default values in JSON schema to reflect values.yaml

* remove defautl templated values

eladkal committed:
Clarify release announcements on social media (#30639)

utkarsharma2 committed:
DynamoDBHook - waiter_path() to consider `resource_type` or `client_type` (#30595)

* Add  while initilizing

* Add  while initilizing

* Add logic to pick either client_type or resource_type

* Add test case

* Assert expected path

bbovenzi committed:
Improve task & run actions ux in grid view (#30373)

* update run clear+mark, update task clear

* add mark as tasks and include list of affected tasks

* Add support for mapped tasks, add shared modal component

* Clean up styling, restore warning for past/future tg clear

maahir22 committed:
Add command to get DAG Details via CLI (#30432)



---------

Co-authored-by: Hussein Awala <hussein@awala.fr>
Co-authored-by: Hussein Awala <houssein.awala.96@gmail.com>

sean-rose committed:
When clearing task instances try to get associated DAGs from database (#29065)

* When clearing task instances try to get associated DAGs from database.

This fixes problems when recursively clearing task instances across multiple DAGs:
  * Task instances in downstream DAGs weren't having their `max_tries` property incremented, which could cause downstream external task sensors in reschedule mode to instantly time out (issue #29049).
  * Task instances in downstream DAGs could have some of their properties overridden by an unrelated task in the upstream DAG if they had the same task ID.

* Use session fixture for new `test_clear_task_instances_without_dag_param` test.

* Use session fixture for new `test_clear_task_instances_in_multiple_dags` test.

---------

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

eladkal committed:
Organize Amazon providers docs index (#30541)

vandonr-amz committed:
preload airflow imports before dag parsing to save time (#30495)



---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

phanikumv committed:
Add delete inactive run functionality to databricks provider (#30646)

rohan472000 committed:
Create audit_logs.rst (#30405)



---------

Co-authored-by: Josh Fell <48934154+josh-fell@users.noreply.github.com>

pierrejeambrun committed:
Present affected task instances as table (#30633)

jedcunningham committed:
Helm chart 1.9.0 has been released (#30649)

ephraimbuddy committed:
Add 2.6.0b1 to issue template (#30652)

ying-w committed:
add missing project_id in BigQueryGetDataOperator (#30651)

potiuk committed:
Properly classify google_vendor package to google provider (#30659)

We've recently added google_venor package to vendor-in ads library,
and we had to do it outside of regular google provider package,
because internally the library assumed our google package is top
level package when discovering the right relative imports (#30544).

This confused the pre-commit that updates provider depedencies to
not recognise the package and print warnings about bad classification.

Special case handling will classify it to google provider.

potiuk committed:
Make pandas optional in workday calendar example (#30660)

The workday calendar expected pandas to be available and it is part
of our examples, however Airflow does not have pandas as a core
dependency, so in case someone does not have pandas installed, importing
of the workday example would fail.

This change makes pandas optional and fallbacks to regular working
days for the example in case it is not available (including warning
about it). It also fixes a slight inefficiency where the
USFederalHoliday calendar has been created every time next workday
was calculated.

VladaZakharova committed:
Update Google Campaign Manager360 operators to use API v4 (#30598)

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>

Bowrna committed:
Merge branch 'patch-3' of https://github.com/Bowrna/airflow
